### ISSUE TITLE
Failed to connect to the camera [Reolink GO]

### ISSUE THREAD

**Issue by LadislavMargai** [QuantumEntangledAndy/neolink#79]
**Describe the bug**
When I try to save an image/turn off the PIR sensor/anything, it will always fail on error `Error: Failed to connect to the camera, check credentials and network`.

**To Reproduce**
1. Configuration file:
````
bind = "0.0.0.0"

[[cameras]]
name = "driveway"
username = "admin"
password = "my_password"
uid = "my_camera_uid"
discovery = "remote"
````
2. Launch Neolink: `neolink --config sample_config.toml image --file-path C:\temp driveway`
3. I tried all the discovery options `local`, `remote`, `map`, `relay`
4. I tried to turn off the firewall, different ISP provider, no success

**Result**
````
[2023-05-03T20:40:56Z INFO  neolink] Neolink 42ff221b0735bf55fbf9ba2508329ef1fdb540e9 release
[2023-05-03T20:40:56Z INFO  neolink::utils] driveway: Connecting to camera at UID: my_camera_uid
Error: Failed to connect to the camera, check credentials and network

Caused by:
    0: Failed to connect to camera driveway at UID: my_camera_uid on channel 0
    1: Cannot contact camera at given address
````

**Expected behavior**
It would connect to the camera and take an image/turn off the PIR/any other action.

**Versions**
NVR software: n/a
Neolink software: [Neolink 0.5.8](https://github.com/QuantumEntangledAndy/neolink/releases/tag/0.5.8), windows 10
Reolink camera model and firmware: Reolink GO, v1.0.213.00

**Comment by QuantumEntangledAndy** [comment]
Is this a cellular? Haven't REd those yet. I believe that there connection method is different. I was originally hoping that the relay mode would help with that since it connects via Reolink but it seems there's something different with its communication. 

If you can provide wireshark dumps of the communications I might be able to figure out what is different With these camera.

**Comment by LadislavMargai** [comment]
Oh, I thought the issues with cellular cameras were already addressed.

**Comment by QuantumEntangledAndy** [comment]
Tracking this in #86. Please do test if you are able

**Comment by Devristo** [comment]
I am not the OP, but it works with my Reolink Go Plus. I only tested the mainStream (which is very choppy) and the subStream. The subStream works great.

**Comment by QuantumEntangledAndy** [comment]
Try decreasing your bitrate a bit to get mainstream to work. Very choppy usually means its too much data for neolink to handle well

**Comment by Devristo** [comment]
That does help, but even on lowest bitrate (1024) it is a bit choppy. But I think it could be 4g reception related, anyway I just found this piece of software half hr ago. Really happy with the result already!

**Comment by QuantumEntangledAndy** [comment]
Really? Well you have good timing then. Cellular has been on the todo list for months. I'm going to work on client pause updates next so that it only connects to sub/main stream when a rtsp cline is connected. Should let us watch the substream constantly and then can just update to main when we need to (frigate does this well I think)

**Comment by QuantumEntangledAndy** [comment]
You can also play with the `buffer_size = 100` to try and smooth it out. The max is 500 so try that and see if it works. It will mean that your stream will be futher behind live though

**Comment by PrplHaz4** [comment]
Seems to work pretty good for me too - nice work @QuantumEntangledAndy !  I know you've been battling this one for a while.

For those looking to test, the docker image is here:
`quantumentangledandy/neolink:feature-cellular`

mainStream performance was better than expected on my medium quality LTE connection.
![image](https://github.com/QuantumEntangledAndy/neolink/assets/841414/3a0466fb-53ff-4451-9a06-1164eebcd84f)

**Comment by QuantumEntangledAndy** [comment]
Sounds good, I'm going to get it merged and release as 0.5.12
